src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/GeneratorTests.Baselines.cs

```diff
@@ -965,5 +965,153 @@ public abstract class AbstractType_CannotInit
 
             Assert.Equal(2, result.Diagnostics.Where(diag => diag.Id == Diagnostics.TypeNotSupported.Id).Count());
         }
+
+        [Fact]
+        public async Task UnsupportedTypes()
+        {
+            string source = """
+                using System;
+                using System.Collections.Generic;
+                using System.Reflection;
+                using System.Runtime.Serialization;
+                using Microsoft.Extensions.Configuration;
+                using Microsoft.Extensions.DependencyInjection;
+
+                public class Program
+                {
+                	public static void Main()
+                	{
+                		ConfigurationBuilder configurationBuilder = new();
+                		IConfiguration configuration = configurationBuilder.Build();
+
+                        // var ws = config.GetSection("MySection");
+                        var w = new Options();
+                        configuration.Bind(w);
+                        configuration.Get<Options>(_ => { });
+
+                        // Should generate a diagnostics
+                        configuration.Get<List<IntPtr>>(_ => { });
+                	}
+                }
+
+                public class Options
+                {
+                    public string? Name { get; set; }
+                    public int Age { get; set; }
+                    public List<string> List { get; set; }
+                    public string[] Array { get; set; }
```

Comment:
Could you also add:

1. A POCO with a phantom type that doesn't get skipped, i.e. for `record Record<T>(int x);` ensure `Record<Action>` works as expected.
2. a non-generic type that explicitly implements a collection type with an unsupported collection element, i.e. `class MyCollection : IDictionary<string, Action>`.